### ISSUE TITLE
support for setting spawn limit via puppet

### ISSUE THREAD

**Issue by cabecada** [sensu/sensu-puppet#727]
we have hit the bottleneck of too many checks (JIT) from a single server leading to issues addressed by https://github.com/sensu/sensu/pull/1265. i know there are other workarounds to deal with this, but since this is public(?) config, i guess it should be configurable via  puppet.

but it seems this is not yet configurable via puppet.
any plans to include this configurable option ?

**Comment by ghoneycutt** [comment]
Thanks @cabecada for reporting this!

@agoddard @cwjohnston I cannot find any documentation for this option on our site. Could you point me in the right direction?

Which file should this go in?

**Comment by cwjohnston** [comment]
You're right there's no documentation for this setting presently. https://github.com/sensu/sensu-docs/issues/451 is tracking that need. Because this modifies the concurrency of both sensu-server pipe handler execution and sensu-client check execution (when running on same system), I recommend using /etc/sensu/conf.d/spawn.json

**Comment by cabecada** [comment]
This is what i tried but i guess this has to go via providers and types?
though hardcoded, but it can be parameterized.

```
class profiles::sensu::spawn {

  $config = {
    "sensu" => {
      "spawn" => {
        "limit" => 20
      }
    }
  }

  file { "/etc/sensu/conf.d/spawn.json":
    mode    => '0644',
    owner   => 'sensu',
    group   => 'sensu',
    require => Package['sensu'],
    content => inline_template(
      '<%= require "json"; (JSON.pretty_generate @config) + $/ %>'
    ),
    notify  => Class['sensu::client::service']
  }
}

Notice: /Stage[main]/Profiles::Sensu::Spawn/File[/etc/sensu/conf.d/spawn.json]/ensure: defined content as '{md5}8ee91eababa124edfe36cc58e637ed32'
Debug: /Stage[main]/Profiles::Sensu::Spawn/File[/etc/sensu/conf.d/spawn.json]: The container Class[Profiles::Sensu::Spawn] will propagate my refresh event
Info: /Stage[main]/Profiles::Sensu::Spawn/File[/etc/sensu/conf.d/spawn.json]: Scheduling refresh of Class[Sensu::Client::Service]

and the sensu-client.log
{"timestamp":"2017-07-10T10:11:57.986676+0000","level":"warn","message":"loading config file","file":"/etc/sensu/conf.d/spawn.json"}
{"timestamp":"2017-07-10T10:11:57.986712+0000","level":"warn","message":"config file applied changes","file":"/etc/sensu/conf.d/spawn.json","changes":{"sensu":{"spawn":{"limit":[12,20]}}}}


```

**Comment by ghoneycutt** [comment]
Thanks everyone! @jeffmccune 's fix is in v2.24.0
